### DISCUSSION TITLE
Get the CustomizedURL

### DISCUSSION THREAD

**Discussion by anmugerm** [amzn/selling-partner-api-models#3647] [General]
I'm trying to get the CustomizedURL of an OrderItem but I'm getting stuck. I've also been looking for an example for the correct call for many hours, but can only find old entries from 2020, where there were obviously problems with the API.
In the old MWS I get the CustomizedURL. I also have all the order data (order, items, address...) together, only the CustomizedURL is missing.

I tried the following:

GET /orders/v0/orders/303-xxxxxx-xxxxxx/orderItems
Here I get the data of the order back:

`
SimpleXMLElement Object
(
    [OrderItems] => SimpleXMLElement Object
        (
            [Order] => SimpleXMLElement Object
                (
                    [ProductInfo] => SimpleXMLElement Object
                        (
                            [NumberOfItems] => 1
                        )

                    [BuyerInfo] => SimpleXMLElement Object
                        (
                        )

                    [ItemTax] => SimpleXMLElement Object
                        (
                            [CurrencyCode] => EUR
                            [Amount] => 2.71
                        )

                    [QuantityShipped] => 1
                    [BuyerRequestedCancel] => SimpleXMLElement Object
                        (
                            [IsBuyerRequestedCancel] => false
                            [BuyerCancelReason] => SimpleXMLElement Object
                                (
                                )

                        )

                    [ItemPrice] => SimpleXMLElement Object
                        (
                            [CurrencyCode] => EUR
                            [Amount] => 16.95
                        )

                    [ASIN] => xxxxxx
                    [SellerSKU] => xx-xxx-xxx
                    [Title] => Wunschtext fuer xxx
                    [IsGift] => false
                    [ConditionSubtypeId] => New
                    [IsTransparency] => SimpleXMLElement Object
                        (
                        )

                    [QuantityOrdered] => 1
                    [PromotionDiscountTax] => SimpleXMLElement Object
                        (
                            [CurrencyCode] => EUR
                            [Amount] => 0.00
                        )

                    [ConditionId] => New
                    [PromotionDiscount] => SimpleXMLElement Object
                        (
                            [CurrencyCode] => EUR
                            [Amount] => 0.00
                        )

                    [OrderItemId] => 301xxxxxxxxxxxxxxx
                )

        )

    [AmazonOrderId] => 303-xxxxxx-xxxxxx
)
`

In part, I understood the documentation to mean that the data should be in the BuyerInfo, but this is empty here.
If I take them with 
GET /orders/v0/orders/303-xxxxxx-xxxxxx/buyerInfo
I only get name and email

I then tried using the special data get with
GET /orders/v0/orders/303-xxxxxx-xxxxxx/orderItem/301xxxxxxxxxxxxxxx/BuyerCustomizedInfoDetail
or 
GET /orders/v0/orders/303-xxxxxx-xxxxxx/orderItem/301xxxxxxxxxxxxxxx/OrderItemBuyerInfo
query, but only get one here: 

`Error: Status code 403: {
  "errors": [
    {
      "code": "Unauthorized",
      "message": "Access to requested resource is denied.",
      "details": ""
    }
  ]
}`

I also received this message during my tests if I did not include missing data such as the MarketplaceIds in my query. But I don't know what else I have to hand over.

Here the questions:
- With which call do I get the CustomizedURL?
- If it should actually be due to rights settings: What else do I have to turn on? I have actually granted all available rights.

**Comment by stefnats** [comment]
Hey there,

so the docs state that you get the `CustomizedURL` with this call:

https://developer-docs.amazon.com/sp-api/docs/orders-api-v0-reference#get-ordersv0ordersorderidorderitemsbuyerinfo

So the call is `GET /orders/v0/orders/{orderId}/orderItems/buyerInfo`

It should be in this object / json path: `Orders.X.ItemBuyerInfo.BuyerCustomizedInfo.CustomizedURL`

The difference between your call is that you requested
`GET /orders/v0/orders/{orderId}/buyerInfo`
but the right call is 
`GET /orders/v0/orders/{orderId}/orderItems/buyerInfo`

Please let me know if this is correct.

**Comment by anmugerm** [comment]
Thanks very much! It is working.
